### PR TITLE
Add ReceiveNever cop

### LINKED ISSUE #607: Cop idea: detect usage of `.never` and suggest `not_to receive` instead
Bad:
`expect(instance).to receive(:method).never`

Good:
`expect(instance).not_to receive(:method)`

### PR DESCRIPTION
**Pull request by Darhazer**
Fixes #607 

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [x] Added the new cop to `config/default.yml`.
* [x] The cop documents examples of good and bad code.
* [x] The tests assert both that bad code is reported and that good code is not reported.

### CODE REVIEW

**Comment by bbatsov** [comment]
Btw, what do you think about putting together some RSpec Style Guide as a reference for the RSpec cops? 

It'd be nice if cops where linked to some document with the rationale behind them.

File: manual/cops_rspec.md [hunk 1]
```diff
@@ -1932,6 +1932,28 @@ expect(foo).to receive(:bar).at_most(:twice).times
...
+# bad
+expect(foo).to receive(:bar).never
+
+# good
+expect(foo).not_to receive(:bar)
+```
+
+### References
+
+* [http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ReceiveNever](http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ReceiveNever)
```
Comment (bbatsov) [inline manual/cops_rspec.md]: This was originally intended to be used for links to blog posts and whatever, complementary to a standard "Style Guide" link.

File: lib/rubocop/cop/rspec/receive_never.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+              location: :selector
+            )
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(node.parent.loc.selector, 'not_to')
+            range = range_between(
+              node.loc.selector.begin_pos - 1, # match the dot as well
```
[the remark this answers is not in the document]
Comment (Darhazer) [reply lib/rubocop/cop/rspec/receive_never.rb]: Fixed

File: lib/rubocop/cop/rspec/receive_never.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        MSG = 'Use `not_to receive` instead of `never`.'.freeze
+
+        def_node_matcher :expectation?, Expectations::ALL.send_pattern
+
+        def_node_search :never_receive, <<-PATTERN
+          (send #expectation? :to
+            {
+              $(send (send nil? :receive ...) :never)
+              $(send (send (send nil? :receive ...) ...) :never)
```
[the remark this answers is not in the document]
Comment (Darhazer) [reply lib/rubocop/cop/rspec/receive_never.rb]: Fixed

**Comment by bquorning** [comment]
@bbatsov Good points both, but not specific to this PR. Would you mind opening two new issues where we can discuss them?

File: manual/cops_rspec.md (hunk 1 above)
Comment (Darhazer) [reply manual/cops_rspec.md]: The rake tasks intercepts "StyleGuide" from the config, as we have full url there :/

File: spec/rubocop/cop/rspec/receive_never_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  it 'flags usage of `never` with `expect_any_instance_of`' do
+    expect_offense(<<-RUBY)
+      expect_any_instance_of(Foo).to receive(:bar).with(baz).never
+                                                             ^^^^^ Use `not_to receive` instead of `never`.
+    RUBY
+  end
+
+  it 'allows method callsed never' do
```
Comment (bquorning) [inline spec/rubocop/cop/rspec/receive_never_spec.rb]: typo: callsed

File: lib/rubocop/cop/rspec/receive_never.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      #
+      #     # good
+      #     expect(foo).not_to receive(:bar)
+      #
+      class ReceiveNever < Cop
+        include RangeHelp
+
+        MSG = 'Use `not_to receive` instead of `never`.'.freeze
+
+        def_node_matcher :expectation?, Expectations::ALL.send_pattern
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/receive_never.rb]: This marcher seems to not be used.

**Comment by Darhazer** [comment]
Updated

**Pull request merged**